release/release.go

```diff
@@ -168,6 +171,36 @@ func getWSLVersion() int {
 	return 2
 }
 
+type snapYaml struct {
+	Name string `yaml:"name"`
+}
+
+// Get the name of the base snap if running inside one.
+func getBaseSnapName() (string, error) {
+	yamlData, err := ioutil.ReadFile("/meta/snap.yaml")
```

Comment:
I wonder if /etc/os-release might be a slightly better place? We could use `ID=ubuntu-core\nVARIANT_ID=ubuntu-core-desktop`  [1] or something similar?  

[1] https://www.freedesktop.org/software/systemd/man/os-release.html

Reply:
The `snap.yaml` check was recommended by @jhenstridge, at least for a first implementation. I think we ultimately need a specific flag, so any future core desktop can easily opt-in to this behaviour.

I think there's three IDs we need, as Ubuntu Core Desktop is:
- A specific OS
- A Core Desktop OS
- A version of Ubuntu Core

I think this would be represented as (might require snapd to be updated to check ID_LIKE):
- `ID=ubuntu-core-desktop\nID_LIKE=ubuntu-core`
- `ID=lubuntu-core-desktop\nID_LIKE=ubuntu-core ubuntu-core-desktop`

Reply:
My thoughts were that it was more important to get the `release.OnCoreDesktop` variable available to unblock other work, with the understanding that we could change how we determine the value of OnCoreDesktop is determined later.

With that said, the use of `VARIANT_ID` does sound like a reasonable fit for what we're trying to do. Looking at the spec, the variant ID is effectively private to the the distro ID, so we could just do `ID=ubuntu-core` and `VARIANT_ID=desktop`.

What we are building mostly behaves like regular Ubuntu Core, so I don't think we'd want to change the distro ID.

Reply:
as I said eleswhere this means that ideally this should also get reflected into ReleaseInfo

Reply:
Changed to using VARIANT_ID